src/components/apis/api-products/ko/runtime/api-products-dropdown.ts

```diff
@@ -0,0 +1,144 @@
+  public readonly selectedProductName: ko.Observable<string>;
+  public readonly selection: ko.Computed<string>;
+
+  constructor(
+    private readonly apiService: ApiService,
+    private readonly router: Router,
+    private readonly routeHelper: RouteHelper
+  ) {
+    this.detailsPageUrl = ko.observable();
+    this.allowSelection = ko.observable(false);
+    this.products = ko.observableArray([]);
+    this.selectedApiName = ko.observable();
+    this.working = ko.observable();
+    this.pattern = ko.observable();
+    this.pageNumber = ko.observable(1);
+    this.totalPages = ko.observable(0);
+    this.selectedProduct = ko.observable();
+    this.selectedProductName = ko.observable();
+    this.selection = ko.computed(() => {
+        const product = ko.unwrap(this.selectedProduct);
+        return product ? product.displayName : "Select API: products";
+    });
+  }
+
+  @Param()
+  public allowSelection: ko.Observable<boolean>;
+  
+  @Param()
+  public detailsPageUrl: ko.Observable<string>;
+
+  @OnMounted()
+  public async initialize(): Promise<void> {
+    const apiName = this.routeHelper.getApiName();
+
+    if (apiName) {
+      this.selectedApiName(apiName);
+      await this.loadPageOfProducts();
+    }
+
+    await this.resetSearch();
+
+    this.pattern
+    .extend({ rateLimit: { timeout: Constants.defaultInputDelayMs, method: "notifyWhenChangesStop" } })
+    .subscribe(this.resetSearch);
+
+    this.router.addRouteChangeListener(this.onRouteChange);
+    this.pageNumber.subscribe(this.loadPageOfProducts);
+  }
+
+  /**
+   * Initiates searching products.
+   */
+   public async resetSearch(): Promise<void> {
+        this.pageNumber(1);
+        this.loadPageOfProducts();
+    }
+
+  private async onRouteChange(): Promise<void> {
+    const apiName = this.routeHelper.getApiName();
+
+    if (
+      !apiName ||
+      (apiName === this.selectedApiName())
+    ) {
+      await this.resetSearch();
+      return;
+    }
+
+    this.selectedApiName(apiName);
+    await this.loadPageOfProducts();
+
+    await this.resetSearch();
+  }
+
+  /**
+   * Loads page of products.
+   */
+  public async loadPageOfProducts(): Promise<void> {
+    try {
+        this.working(true);
+
+        const pageNumber = this.pageNumber() - 1;
+
+        const query: SearchQuery = {
+            pattern: this.pattern(),
+            skip: pageNumber * Constants.defaultPageSize,
+            take: Constants.defaultPageSize
+        };
+      
+      const apiName = this.selectedApiName();
+
+      const pageOfProducts = await this.apiService.getApiProductsPage(apiName, query);
+
+      this.products(pageOfProducts.value);
+      this.totalPages(Math.ceil(pageOfProducts.count / Constants.defaultPageSize));
+
+      const productName = this.routeHelper.getProductName();
+
+        this.selectedProduct(productName
+            ? pageOfProducts.value.find(item => item.id.endsWith(productName))
+            : pageOfProducts.value[-1]);
```

Comment:
What does it mean `pageOfProducts.value[-1]` ? if value is an array which I suppose it is, then it would return `undefined`. Why not just return write undefined?

On the other hand, If undefined do not need to be set to ko.Observable, then you can do it like this
```suggestion
            if(productName){
                this.selectedProduct(pageOfProducts.value.find(item => item.id.endsWith(productName)));
             }
```
Or in one line
```suggestion
            productName && this.selectedProduct(pageOfProducts.value.find(item => item.id.endsWith(productName)));
```

Reply:
Agreed. Thanks for the heads up.